src/plugins/cachefilter/cachefilter.c

```diff
@@ -0,0 +1,73 @@
+/**
+ * @file
+ *
+ * @brief Source for cachefilter plugin
+ *
+ * @copyright BSD License (see doc/COPYING or http://www.libelektra.org)
+ *
+ */
+
+#include "cachefilter.h"
+
+#include <kdbhelper.h>
+
+
+int elektraCachefilterGet (Plugin * handle, KeySet * returned, Key * parentKey)
+{
+	if (!elektraStrCmp (keyName (parentKey), "system/elektra/modules/cachefilter"))
+	{
+		KeySet * contract = ksNew (
+			30, keyNew ("system/elektra/modules/cachefilter", KEY_VALUE, "cachefilter plugin waits for your orders", KEY_END),
+			keyNew ("system/elektra/modules/cachefilter/exports", KEY_END),
+			keyNew ("system/elektra/modules/cachefilter/exports/get", KEY_FUNC, elektraCachefilterGet, KEY_END),
+			keyNew ("system/elektra/modules/cachefilter/exports/set", KEY_FUNC, elektraCachefilterSet, KEY_END),
+#include ELEKTRA_README (cachefilter)
+			keyNew ("system/elektra/modules/cachefilter/infos/version", KEY_VALUE, PLUGINVERSION, KEY_END), KS_END);
+		ksAppend (returned, contract);
+		ksDel (contract);
+
+		return 1; // success
+	}
+
+	// filter keys that need to be cached
+	KeySet * temp = 0;
+	KeySet * toBeCached = ksNew (ksGetSize (returned), KS_END);
```

Comment:
Yes, calloc is used and it would not be a good idea to return uninialized memory, so its a guarantee. I updated the docu in 5b0c21f9369404ee5eedf67ddd719db276991759